src/main/java/com/novauc/HibernateQueryMethodsController.java

```diff
@@ -0,0 +1,73 @@
+package com.novauc;
+
+import com.sun.xml.internal.ws.policy.privateutil.PolicyUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Created by Merlin on 3/16/17.
+ */
+@Controller
+public class HibernateQueryMethodsController {
+    @Autowired
+    CustomerRepository customers;
+
+    @Autowired
+    PurchaseRepository purchases;
+
+//    @RequestMapping(path = "/", method = RequestMethod.GET)
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model, String category) {
+        List<Purchase> purchaseList = (ArrayList) purchases.findAll();
+        if (category != null) {
+            purchaseList = (List) purchases.findByCategory(category);
+        } else {
+            purchaseList = (List) purchases.findAll();
+        }
+        model.addAttribute("purchases", purchaseList);
+        return "home";
+    }
+
```

Comment:
Nice job ross.  For the second part of the assignment(adding paging and sorting) you won't need a new repository like you have started to setup.  You just need to have one of your current repositories extend something differently.